pkg/content/validator.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package content
+
+import (
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+var _ swarm.ChunkValidator = (*ContentAddressValidator)(nil)
+
+// ContentAddressValidator validates that the address of a given chunk
+// is the content address of its contents.
+type ContentAddressValidator struct {
```

Comment:
Same as the above, you can ommit `Content` in naming inside `content` package if you like.